src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesReconService.kt

```diff
@@ -45,17 +48,40 @@ class ActivitiesReconService(
         val dpsBookingCounts = async { activitiesApiService.getAllocationReconciliation(prisonId) }
         val compareResults = compareBookingCounts(nomisBookingCounts.await(), dpsBookingCounts.await())
 
-        publishTelemetry(prisonId, compareResults)
+        publishTelemetry("allocation", prisonId, compareResults)
       }
     } catch (e: Exception) {
       log.error("Allocation reconciliation report failed for prison $prisonId", e)
       telemetryClient.trackEvent("activity-allocation-reconciliation-report-error", mapOf("prison" to prisonId))
     }
 
-  private suspend fun compareBookingCounts(nomisResults: NomisResponse, dpsResults: DpsResponse): CompareBookingsResult {
+  suspend fun attendancesReconciliationReport(prisonId: String, date: LocalDate) =
+    try {
+      coroutineScope {
+        val nomisBookingCounts = async { nomisApiService.getAttendanceReconciliation(prisonId, date) }
+        val dpsBookingCounts = async { activitiesApiService.getAttendanceReconciliation(prisonId, date) }
+        val compareResults = compareBookingCounts(nomisBookingCounts.await(), dpsBookingCounts.await())
+
+        publishTelemetry("attendance", prisonId, compareResults, date)
+      }
+    } catch (e: Exception) {
+      log.error("Attendance reconciliation report failed for prison $prisonId", e)
+      telemetryClient.trackEvent("activity-attendance-reconciliation-report-error", mapOf("prison" to prisonId, "date" to "$date"))
+    }
```

Comment:
The main change is to implement the attendance reconciliation report - not plumbed into an endpoint yet.